sdk/cargo-build-sbf/src/main.rs

```diff
@@ -1048,6 +1066,13 @@ fn main() {
                 .takes_value(false)
```

Comment:
I can't comment higher up, but `let rust_base_version = get_base_rust_version(...)` on line 995 will use homedir to find the tools version. Since this is just for logging with CI caches, I assume it's OK for it to fail and return an empty string, but I wanted to flag it either way.

Reply:
in nix wrapper and override I set exact version of tools and path it to cargo-build-sbf. 

if that is okey return empty string will do it, so

Reply:
```rust
    let rust_base_version = get_base_rust_version(DEFAULT_PLATFORM_TOOLS_VERSION);
    let version = format!(
        "{}\nplatform-tools {}\n{}",
        crate_version!(),
        DEFAULT_PLATFORM_TOOLS_VERSION,
        rust_base_version,
    );
    let matches = clap::Command::new(crate_name!())
        .about(crate_description!())
        .version(version.as_str())
```
oh, recalled what problem was.

1. I can invoke cargo build sbf in nix even if this lines runs. no worries.
2. I cannot not invoke it in case of nix, because cli parameters not yet parsed. until i change usage of clap and provide own version command.  i want to avoid doing anything with next in this PR `.version(version.as_str())`

but seems solution which will be backward compatible and provide correct tools and rust version is to do custom command and resolve 

```sh
bash-5.1$ cargo run -- --version
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.17s
     Running `/home/dz/github.com/anza-xyz/agave/target/debug/cargo-build-sbf --version`
solana-cargo-build-sbf 2.2.0
platform-tools v1.43

bash-5.1$ 
```  

and doing `--version --WHATEVER_PARAMETERS_OVERRIDERUST` as option.

Reply:
Ok great, just wanted to make sure everything is good for your use case